server/enhancedDate.js

```diff
@@ -1 +1,78 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+
+var enhancedDate = function () {
+	
+	var publicAPI={};
+	
+	var privateAPI ={
+		days : ['Sunday', 'Monday', 'Tuesday', 'Wednesday','Thursday', 'Friday', 'Saturday'],
+		months : ['January', 'February', 'March', 'April', 'May','June', 'July', 'August', 'September', 'October','November', 'December']
+	};
+	
+	var newDate;
+	var now = new Date();
+	var setNewDate = function(){
+		if (!newDate){
+			newDate= new Date();			
+		} 
+	};
+	
+    function setDate(date) {
+    	if (date){
+    		if (typeof date === "number"){
+    			newDate = new Date(date);
+    		} else if(date instanceof Date){
+    			newDate= date;
```

Comment:
When date is passed in as an object, it's passed by reference. This means that if the date object is modified outside the module, it will modify the date stored internally as well. Better to instantiate a new object based on the object passed in: `newDate = newDate(date);`